### PR TITLE
[Feature]: Add csv table directive

### PR DESCRIPTION
**Pull request by jaapio**
Add csv table directive, this allows users to import a csv as table.

### CODE REVIEW

**Review by linawolf — APPROVED** [review]
I would suggest to maybe add a few more complex test cases

**Comment by linawolf** [comment]
Also note that there is such a feature in rst: https://docutils.sourceforge.io/docs/ref/rst/directives.html#csv-table-1, we also use it in TYPO3 https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/WritingReST/Reference/Content/Tables.html#csv-table

File: tests/Integration/tests/table-csv-table-with-markup/input/test.csv [hunk 1]
```diff
@@ -0,0 +1,4 @@
+col 1, col 2
+**a**, "quoted column is unquoted"
```
Comment (linawolf) [inline tests/Integration/tests/table-csv-table-with-markup/input/test.csv:2]: I am not sure this makes sense in most use cases... Often csv files would come from others sources so I would not really expect rst markup to be applied? What you think?

Comment (jaapio) [reply tests/Integration/tests/table-csv-table-with-markup/input/test.csv:2]: According to this page: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/tables.html#csv-files

it is supported by sphinx?

Comment (linawolf) [reply tests/Integration/tests/table-csv-table-with-markup/input/test.csv:2]: Ok, didnt know that

**Pull request merged**